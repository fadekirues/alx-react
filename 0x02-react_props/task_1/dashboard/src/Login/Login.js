import React,{useRef} from "react";
import "./Login.css";

function Login() {
  const emailRef = useRef(null);
  const passRef = useRef(null);

  const handleLabelClick = (inputRef) => {
    inputRef.current.focus();
  };

  return (
    <div>
      <div>
        <label htmlFor="email" onClick={() => handleLabelClick(emailRef)}>
          Email:
        </label>
        <input type="email" id="email" ref={emailRef} />
      </div>
      <div>
        <label htmlFor="password" onClick={() => handleLabelClick(passRef)}>
          Password:
        </label>
        <input type="password" id="password" ref={passRef} />
      </div>
      <button>OK</button>
    </div>
  );
}

export default Login;
