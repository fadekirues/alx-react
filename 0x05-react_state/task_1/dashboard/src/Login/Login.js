import React, { useRef, useState, useEffect } from "react";
import { StyleSheet, css } from "aphrodite";

const styles = StyleSheet.create({
  loginContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    padding: "20px",
  },
  inputContainer: {
    marginBottom: "20px",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  label: {
    marginRight: "10px",
    cursor: "pointer",
  },
  input: {
    padding: "5px",
    border: "1px solid #ccc",
    borderRadius: "4px",
  },
  button: {
    padding: "10px 20px",
    backgroundColor: "#007bff",
    color: "white",
    border: "none",
    borderRadius: "4px",
    cursor: "pointer",
  },
});

function Login() {
  const emailRef = useRef(null);
  const passRef = useRef(null);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [enableSubmit, setEnableSubmit] = useState(false);

  const handleLabelClick = (inputRef) => {
    inputRef.current.focus();
  };

  const handleChangeEmail = (e) => {
    setEmail(e.target.value);
  };

  const handleChangePassword = (e) => {
    setPassword(e.target.value);
  };

  const handleLoginSubmit = (e) => {
    e.preventDefault();
    setIsLoggedIn(true);
  };

  // Enable the submit button when both email and password are not empty
  useEffect(() => {
    setEnableSubmit(email.trim() !== "" && password.trim() !== "");
  }, [email, password]);

  return (
    <div className={css(styles.loginContainer)}>
      <form onSubmit={handleLoginSubmit}>
        <div className={css(styles.inputContainer)}>
          <label
            htmlFor="email"
            className={css(styles.label)}
            onClick={() => handleLabelClick(emailRef)}
          >
            Email:
          </label>
          <input
            type="email"
            id="email"
            ref={emailRef}
            className={css(styles.input)}
            value={email}
            onChange={handleChangeEmail}
          />
        </div>
        <div className={css(styles.inputContainer)}>
          <label
            htmlFor="password"
            className={css(styles.label)}
            onClick={() => handleLabelClick(passRef)}
          >
            Password:
          </label>
          <input
            type="password"
            id="password"
            ref={passRef}
            className={css(styles.input)}
            
            value={password}
            onChange={handleChangePassword}
          />
        </div>
        <input
          type="submit"
          value="OK"
          className={css(styles.button)}
          disabled={!enableSubmit}
        />
      </form>
    </div>
  );
}

export default Login;
