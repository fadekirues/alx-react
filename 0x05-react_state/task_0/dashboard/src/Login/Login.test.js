import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Login from "./Login";

// Import react-test-renderer for capturing rendered output
import {create} from "react-test-renderer";

// Configure enzyme with the React adapter
configure({ adapter: new Adapter() });

// Set up jsdom to create a virtual DOM
const { JSDOM } = require("jsdom");
const { window } = new JSDOM("<!doctype html><html><body></body></html>");
global.window = window;
global.document = window.document;

describe("Login Component", () => {
  it("should render without crashing", () => {
    shallow(<Login />);
  });

  it("should render 2 input tags", () => {
    const wrapper = shallow(<Login />);
    const inputElements = wrapper.find("input");
    expect(inputElements).toHaveLength(2);
  });

  it("should render 2 label tags", () => {
    const wrapper = shallow(<Login />);
    const labelElements = wrapper.find("label");
    expect(labelElements).toHaveLength(2);
  });

  it("should match the snapshot", () => {
    const tree = create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
