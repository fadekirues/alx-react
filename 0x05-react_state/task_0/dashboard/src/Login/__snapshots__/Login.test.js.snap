// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match the snapshot 1`] = `
<div
  className="loginContainer_r7txk7"
>
  <div
    className="inputContainer_7u2k1t"
  >
    <label
      className="label_y5lhqh"
      htmlFor="email"
      onClick={[Function]}
    >
      Email:
    </label>
    <input
      className="input_1vpfudx"
      id="email"
      type="email"
    />
  </div>
  <div
    className="inputContainer_7u2k1t"
  >
    <label
      className="label_y5lhqh"
      htmlFor="password"
      onClick={[Function]}
    >
      Password:
    </label>
    <input
      className="input_1vpfudx"
      id="password"
      type="password"
    />
  </div>
  <button
    className="button_1bw7bod"
  >
    OK
  </button>
</div>
`;
